services/llgen/kilda/llgen/protocol.py

```diff
@@ -0,0 +1,334 @@
+#
+
+import collections
+import json
+import socket
+import inspect
+
+
```

Comment:
Need some comments on this file .. how the classes are used .. how it comes together .. examples of how it is used .. etc.